metabase/metabase
# Support both name & field ref-based column keys in viz settings on read and upgrade on write

I realized that we cannot fully migrate `column_settings` because 10-20% of all cards that we know about don't have `result_metadata` and also don't have `table.columns` (that can be used to substitute `result_metadata` in some cases). So we would have to do it the hard way - by supporting both legacy and modern keys on read, and upgrading legacy keys on write.

How to verify:
- On `master` -> New -> Question -> Orders -> Visualize
- Use viz settings to change column titles for multiple columns -> Save
- Export this question and make sure that columns have correct titles in exported files 
- Checkout this branch and open the question
- All column titles should not be lost
- Export should work as before
- Change some column title and save. Check the network tab and see that column keys were rewritten.
- Column titles for unchanged columns should not be lost.
- Export should work as before for this question after saving.
